### ISSUE TITLE
Consider deprecating old Leaflet versions docs, and automate publishing

### ISSUE THREAD

**Issue by mourner** [Leaflet/Leaflet#7712]
Currently dealing with Leaflet docs causes quite a bit of confusion:

- Since `reference.html` is built locally and committed to the repo, we often get PRs on the generated files rather than sources for docs fixes.
- Updating docs for a release is a [quite involved](https://github.com/Leaflet/Leaflet/blob/master/RELEASE.md) manual process — ideally this would be completely automated.
- Old version docs might show up in Google search results instead of latest docs.
- Keeping generated files in the repo (and we have quite a few, including old versions) is a bad practice.

We introduced links to old Leaflet docs back in 2012, when Leaflet was v0.3. Now that it's way more stable and mature and breaking changes are extremely rare, I think we can drop the old versions and only have one reference docs reflecting the latest version.

Also, now that we're using GitHub Actions, automating the docs publishing when tagging a release seems quite doable. cc @jonkoops @IvanSanchez

**Comment by Falke-Design** [comment]
So you want to remove the old docs from the website too? https://leafletjs.com/reference-versions.html
For me it was / is a good way to determine differences between versions and help others if they have problems with plugins and are not on the newest Version

**Comment by mourner** [comment]
@Falke-Design yes. Can you expand on how exactly you'd determine differences this way? The reference pages look identical, with the same sections, and differences so tiny you'd have to know exactly where to look to spot them. IMO the only reliable way to determine differences between versions is the Changelog, and I see no purpose in hosting many nearly-identical docs pages at this stage of the project.

**Comment by Falke-Design** [comment]
I don't know what I was searching but a user of my library was having a issue and then we found out that the function that I used was not existing in 1.6 which he used because of the IOS double tab problem. But yes I searched exactly for the function in the old reference

**Comment by mourner** [comment]
@Falke-Design you could have found it by searching the Changelog as well (and quickly finding the first version where it was added). I don't say that older Leaflet reference docs can't be useful, just that their usefulness doesn't justify the issues I listed above.

**Comment by jonkoops** [comment]
Yeah, I agree that this should be automated, let's consider the publishing of the package separate from the documentation for now so we can chunk up the work more easily. We could use the [release](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#release) event and hook into Github releases to publish things to NPM at the same time.

**Comment by mourner** [comment]
Yep, either a `release` event or a git tag that fits semver.

**Comment by jonkoops** [comment]
Added a change to publish to NPM here https://github.com/Leaflet/Leaflet/pull/7854

**Comment by Malvoz** [comment]
There are snapshots of all the reference versions in the Internet Archive's Wayback Machine:

- [0.7.x](https://web.archive.org/web/20211103212702/https://leafletjs.com/reference-0.7.7.html)
- [1.0.3](https://web.archive.org/web/20211129173330/https://leafletjs.com/reference-1.0.3.html)
- [1.1.0](https://web.archive.org/web/20211103195233/https://leafletjs.com/reference-1.1.0.html)
- [1.2.0](https://web.archive.org/web/20211107001416/https://leafletjs.com/reference-1.2.0.html)
- [1.3.4](https://web.archive.org/web/20220512035219/https://leafletjs.com/reference-1.3.4.html)
- [1.4.0](https://web.archive.org/web/20211030034406/https://leafletjs.com/reference-1.4.0.html)
- [1.5.x](https://web.archive.org/web/20201202155513/https://leafletjs.com/reference-1.5.1.html)
- [1.6.0](https://web.archive.org/web/20211123172545/https://leafletjs.com/reference-1.6.0.html)
- [1.7.1](https://web.archive.org/web/20220505205646/https://leafletjs.com/reference-1.7.1.html)

+1 for deprecating old version docs.

**Comment by Malvoz** [comment]
Should `reference-versions.html` be removed as well? Or do we replace the links in that document with the web archive links above?

**Comment by Falke-Design** [comment]
I think we should replace it web archiv links or at least suggest that the versions can be found in the web archiv

**Comment by Falke-Design** [comment]
Yes you are right
